CHANGELOG.md

```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- Fix incorrectly instantiating `php-http/curl-client` causing argument errors (#1052)
```

Comment:
What about `Fix typehint errors while instantiating the Httplug cURL client by forcing the usage of PSR-17 complaint factories`?

Reply:
That actually sounds pretty good, adjusted!